src/command/exec.rs

```diff
@@ -0,0 +1,44 @@
+use super::{Input, Parser, ParserBase};
+
+#[derive(Clone)]
+pub struct Exec<'a, P: Parser, C> {
+    pub(super) parser: P,
+    pub(super) command: fn(&'a mut C, P::Extract) -> ()
+}
+
+impl<'a, P, C> ParserBase for Exec<'a, P, C>
+where
+    P: Parser,
+    C: 'a,
+    P::Extract: 'static,
+{
+    type Extract = (Command<'a, P::Extract, C>,);
+
+    fn parse<'i>(&self, input: &mut Input<'i>) -> Option<Self::Extract> {
+        let ex = self.parser.parse(input)?;
+        if input.is_empty() {
+            Some((Command {
+                extracted: ex,
+                command: self.command,
+            },))
+        } else {
+            None
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct Command<'a, E, C> {
+    pub(super) extracted: E,
+    pub(super) command: fn(&'a mut C, E) -> ()
```

Comment:
It should be possible to change the type of field `command` to
```rust
for<'a> fn(&'a mut C, E) -> ()
```
 
This would allow removing the `'a` lifetime from the struct.

(The same would have to be done for `Exec`, `And`, etc.)